cypress/e2e/settings.spec.ts

```diff
@@ -245,6 +245,130 @@ describe("Settings page", () => {
     cy.get("@showReach").should("not.be.checked")
   })
 
+  it("Should toggle Privatise staging site button and automatically generate a password", () => {
+    // Arrange
+    // NOTE: Initial state is privatise staging off
+    cy.contains("Privatise staging site")
+      .parent()
+      .parent()
+      .find("input")
+      .as("privatiseStaging")
+    cy.get("@privatiseStaging").should("not.be.checked")
+
+    // Act
+    // Trigger click event
+    // NOTE: We have to force as chakra uses an invisible input as a checkbox
+    cy.get("@privatiseStaging")
+      .check({ force: true, timeout: 0 })
+      .then(() => {
+        // Checking within then because of wonky focus preventing the field from being populated
+        cy.contains("You should change this every 90 days")
+          .parent()
+          .next()
+          .find("input")
+          .as("passwordInput")
+        cy.wait(1000)
+        cy.get("@passwordInput").should("have.length.gt", 0)
+        cy.saveSettings()
+      })
+
+    // Assert
+    cy.get("@privatiseStaging").should("be.checked")
+    cy.contains("Username").parent().next().as("usernameInput")
+    cy.get("@usernameInput").should("be.disabled")
+  })
+
+  it("Should regenerate a password if password field is emptied and privatise button toggled again", () => {
+    // Arrange
+    cy.contains("Privatise staging site")
+      .parent()
+      .parent()
+      .find("input")
+      .as("privatiseStaging")
+    cy.get("@privatiseStaging").should("be.checked")
```

Comment:
this has implicit dep on prev test so if the prev one fail, this will fail too ._.